validate/validate.h

```diff
@@ -68,6 +76,49 @@ static inline bool Contains(const string& search_in, const string& to_find)
   return search_in.find(to_find) != string::npos;
 }
 
+static inline bool IsIpv4(const string& to_validate) {
+	struct sockaddr_in sa;
+	return !(inet_pton(AF_INET, to_validate.c_str(), &sa.sin_addr) < 1);
+}
+
+static inline bool IsIpv6(const string& to_validate) {
+  struct sockaddr_in6 sa_six;
+	return !(inet_pton(AF_INET6, to_validate.c_str(), &sa_six.sin6_addr) < 1);
+}
+
+static inline bool IsIp(const string& to_validate) {
+  return (IsIpv4(to_validate) || IsIpv6(to_validate));
```

Comment:
nit: the extra parentheses are unnecessary.

Reply:
Done.